src/ripple/overlay/impl/PeerImp.cpp

```diff
@@ -2460,25 +2461,15 @@ PeerImp::checkPropose(
         return;
     }
 
+    bool relay;
+
     if (isTrusted)
-    {
-        app_.getOPs().processTrustedProposal(peerPos, packet);
-    }
+        relay = app_.getOPs().processTrustedProposal(peerPos);
```

Comment:
`processTrustedProposal` eventually calls `Consensus<Adaptor>::peerProposalInternal` and checks things like `if (phase_ == ConsensusPhase::accepted)`. That means this proposal is too late for me. At this point, most likely the network doesn't need it anymore. But someone might. I think either way is fine. Just wanted to point it out. 

Also this part seems asymmetric. Ie. if RELAY_UNTRUSTED_PROPOSALS is "all", untrusted proposals get relayed no matter what, but trusted proposals must go through additional checks.

Reply:
Good eyes. Yes, "all" in this case doesn't _really_ mean "all" because of this additional filtering that we perform. But as you note, proposals are, generally, not needed if the corresponding _validation_ is seen.